### PR TITLE
🥳 Update Python module dependencies

### PR DESCRIPTION
**Pull request by dodobas**
- boto3 - https://github.com/boto/boto3/blob/develop/CHANGELOG.rst
- Django - https://github.com/django/django/blob/master/docs/releases/2.1.6.txt
- twilio - https://github.com/twilio/twilio-python/releases/tag/6.24.0
- pip-tools - https://github.com/jazzband/pip-tools/releases
- psycopg2 - https://github.com/psycopg/psycopg2/releases
- flake8 - http://flake8.pycqa.org/en/latest/release-notes/
- smartmin - https://github.com/nyaruka/smartmin/blob/master/CHANGES.md
- stripe - https://github.com/stripe/stripe-python/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by nicpottier** [comment]
This is great and we should get this across the line, but I don't think we want monthly updates.

My proposal would be:
 * Django security updates, immediately
 * Patch versions quarterly (May 1, Aug 1, Nov 1)
 * Major versions yearly (Feb 1)

@rowanseymour @ericnewcomer @norkans7 thoughts?

**Comment by nicpottier** [comment]
And really I could probably be convinced that minor should be yearly and major should be only as needed / required. There is often a lot of useless toil in changing majors.

**Comment by rowanseymour** [comment]
Think my preference would be...

* Dependencies we develop (e.g. smartmin) should be upgraded immediately as the person who just made the changes will know in that moment what if anything needs to be changed in RP
* Any security patch or bug fix (that affects us) upgraded immediately
* Complete dependency refresh every year.. sometimes its best to just upgrade everything together and bug bash... do think there's value in not getting too far behind.

**Comment by dodobas** [comment]
schedule is very simple ... every time Django releases a security update (monthly/bi-monthly), spend a hour or two and check which modules could be updated and what we need to do to use the new version ... if there are create issues for those modules

it's much easier to traverse few updated changelogs, than read through wall of text in a year 

IMHO any other elaborate scheduled module update policy is just going to fail, because it needs to be enforced, and we don't have enforcers on our team :)

**Comment by norkans7** [comment]
I think anything that is really quick(an hour or less) should fine only the updates that need the special attention can be done on schedules

**Comment by nicpottier** [comment]
It isn't a matter of the time involved in the updates, it is a matter that updates tend to break things. We don't want to be in a constant state of change due to updates. Security fixes for Django are one thing, both because they are required and because that project tends to be pretty good about breaking changes. In general we haven't encountered the same with other projects vis-a-vis subtle changes that break shit.

In short, there's only downside to updating libraries when they aren't addressing a bug fix or security issue. Stability is a good thing, if a library version does what we need, there's no need to fuck with it.

**Comment by dodobas** [comment]
And that would be fine if every package (other than Django) had a sensible policy of supporting multiple versions of software, and paid maintainers to backport security updates ... but that is not the case

If we want to apply security updates, then we need to live on the edge, which brings new features and can break stuff.

**Comment by ericnewcomer** [comment]
..and that hits the nail on the head.

We only take forced updates if a security fix forces us too. The risk to stability should be compressed into one period. There is no benefit to spreading out that risk so that we are always in a state of flux which can result in a steadily poor experience for our users. I feel strongly about the schedule @nicpottier suggests. I don't think we should do any piecemeal updates unless a) we have to for security reasons or b) we decide there is some significant benefit that warrants the potential instability. In the latter case where we break the schedule, then it needs to be justified to the team.

I do feel a schedule is a great idea and something we should force on ourselves to make sure we don't fall too far behind -- we've mostly already done this, we just haven't codified it. We have the people on the team for that, so I don't get that criticism.
